api/Controllers/HotelsController.cs

```diff
@@ -85,8 +93,34 @@ public async Task<IActionResult> GetHotelCards([FromQuery] PageParameters pagePa
 
         [Route("count")]
         [HttpGet]
-        public async Task<IActionResult> GetHotelsCount() =>
-            Ok(await _hotelsDb.Hotels.CountAsync());
+        public async Task<IActionResult> GetHotelsCount([FromQuery] FilterParameters filterParams)
+        {
+            var hotelCards = _hotelsDb.Hotels.AsQueryable();
+            if (!string.IsNullOrEmpty(filterParams.Name))
+            {
+                hotelCards = hotelCards.Where(h => h.Name.ToLower().Contains(filterParams.Name.ToLower()));
+            }
+            var hotelCardsNumber = await hotelCards.CountAsync();
+            return Ok(hotelCardsNumber);
+        }
+
+        [Route("names")]
+        [HttpGet]
+        public async Task<IActionResult> GetHotelNames([FromQuery] string name, [FromQuery] string number = "2")
```

Comment:
Is it possible to avoid use of number as string (to avoid `Convert.ToInt32`)

Reply:
It should be possible, gonna investigate and fix it